apps/dashboard/src/components/conditions-editor/conditions-editor-context.tsx

```diff
@@ -11,23 +12,33 @@ export const ConditionsEditorContext = createContext<ConditionsEditorContextType
   getParentGroup: () => null,
 });
 
-export function ConditionsEditorProvider({ children }: { children: React.ReactNode }) {
-  const [query, setQuery] = useState<RuleGroupType>({ combinator: 'and', rules: [] });
-
+export function ConditionsEditorProvider({
+  children,
+  query,
+  onQueryChange,
+}: {
+  children: React.ReactNode;
+  query: RuleGroupType;
+  onQueryChange: (query: RuleGroupType) => void;
+}) {
+  const queryRef = useDataRef(query);
+  const queryChangeRef = useDataRef(onQueryChange);
```

Comment:
the `query` is now passed as a prop, to improve rendering perf we need to wrap in a `useRef` otherwise callbacks will be recreated each time